gensim/models/base_any2vec.py

```diff
@@ -0,0 +1,792 @@
+    """Class for storing and initializing/updating the trainable weights of a model. Also includes
+    tables required for training weights. """
+    def __init__(self, vector_size=None, seed=1):
+        self.vectors = []
+        self.vector_size = int(vector_size)
+        self.seed = seed
+
+    def prepare_weights(self, update=False, vocabulary=None):
+        raise NotImplementedError
+
+    def reset_weights(self, vocab):
+        """Reset all trainable weights to an initial (untrained) state, but keep the existing vocabulary."""
+        raise NotImplementedError
+
+    def update_weights(self, vocab):
+        """
+        Copy all the existing weights, and reset the weights for the newly
+        added vocabulary.
+        """
+        raise NotImplementedError
+
+    def seeded_vector(self, seed_string):
+        """Create one 'random' vector (but deterministic by seed_string)"""
+        raise NotImplementedError
+
+
+class BaseWordEmbedddingsModel(BaseAny2VecModel):
+    def __init__(self, sentences=None, workers=3, vector_size=100, epochs=5, callbacks=(), batch_words=10000,
+                 trim_rule=None, sg=0, alpha=0.025, window=5, seed=1, hs=0, negative=5, cbow_mean=1,
+                 min_alpha=0.0001, compute_loss=False, **kwargs):
+        self.sg = int(sg)
+        if vector_size % 4 != 0:
+            logger.warning("consider setting layer size to a multiple of 4 for greater performance")
+        self.alpha = float(alpha)
+        self.window = int(window)
+        self.random = random.RandomState(seed)
+        self.min_alpha = float(min_alpha)
+        self.hs = hs
+        self.negative = negative
+        self.cbow_mean = int(cbow_mean)
+        self.compute_loss = compute_loss
+        self.running_training_loss = 0
+        self.min_alpha_yet_reached = float(alpha)
+
+        super(BaseWordEmbedddingsModel, self).__init__(
+            workers=workers, vector_size=vector_size, epochs=epochs, callbacks=callbacks, batch_words=batch_words)
+        if sentences is not None:
+            if isinstance(sentences, GeneratorType):
+                raise TypeError("You can't pass a generator as the sentences argument. Try an iterator.")
+            self.build_vocab(sentences, trim_rule=trim_rule)
+            self.train(
+                sentences, total_examples=self.corpus_count, epochs=self.epochs, start_alpha=self.alpha,
+                end_alpha=self.min_alpha, compute_loss=compute_loss)
+        else:
+            if trim_rule is not None:
+                logger.warning(
+                    "The rule, if given, is only used to prune vocabulary during build_vocab() "
+                    "and is not stored as part of the model. Model initialized without sentences. "
+                    "trim_rule provided, if any, will be ignored.")
+
+    # for backward compatibility (aliases pointing to corresponding variables in trainables, vocabulary)
+    @property
+    def iter(self):
```

Comment:
Probably, add "deprecated" decorator for all of this properties (I think we should remove it later).